### PR TITLE
Add kubebuilder image build

### PR DESCRIPTION
**Pull request by yashsingh74**
Added the kubebuilder image build job.

Issue Ref: https://github.com/kubernetes-sigs/kubebuilder/issues/3230

### CODE REVIEW

File: config/jobs/image-pushing/k8s-staging-kubebuilder.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+postsubmits:
+
+  kubernetes-sigs/kubebuilder:
+    - name: post-kubebuilder-push-images
+      cluster: k8s-infra-prow-build-trusted
+      annotations:
+        testgrid-dashboards: sig-api-machinery-kubebuilder
+      decorate: true
+      branches:
+        - ^master$
```
Comment (camilamacedo86) [inline config/jobs/image-pushing/k8s-staging-kubebuilder.yaml:10]: Why we need to use master?

File: config/jobs/image-pushing/k8s-staging-kubebuilder.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        - ^kube-rbac-proxy-releases$
+      spec:
+        serviceAccountName: gcb-builder
+        containers:
+          - image: gcr.io/k8s-staging-test-infra/image-builder:v20230111-cd1b3caf9c
+            command:
+              - /run.sh
```
Comment (camilamacedo86) [inline config/jobs/image-pushing/k8s-staging-kubebuilder.yaml:17]: It should build the proxy image right?

To build today we call: https://github.com/kubernetes-sigs/kubebuilder/blob/kube-rbac-proxy-releases/build/cloudbuild_kube-rbac-proxy.yaml

Which will call https://github.com/kubernetes-sigs/kubebuilder/blob/kube-rbac-proxy-releases/build/build.sh passing the tag version to do the job.

So, what this run.sh does?
What is the expected outcome when this PR get merged?

File: config/jobs/image-pushing/k8s-staging-kubebuilder.yaml (hunk 1 above)
Comment (yashsingh74) [reply config/jobs/image-pushing/k8s-staging-kubebuilder.yaml:10]: Ohh it is not required.
Also, do we need to check against the Tag like mentioned here -https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-ingress-nginx.yaml#L10

File: config/jobs/image-pushing/k8s-staging-kubebuilder.yaml (hunk 2 above)
Comment (yashsingh74) [reply config/jobs/image-pushing/k8s-staging-kubebuilder.yaml:17]: I thought the run.sh will run initially which is present inside the image-builder.

File: config/jobs/image-pushing/k8s-staging-kubebuilder.yaml (hunk 1 above)
Comment (varshaprasad96) [reply config/jobs/image-pushing/k8s-staging-kubebuilder.yaml:10]: Yes it should be just against the tag. I don't think any changes in the branch need to trigger. @camilamacedo86 could you confirm?

File: config/jobs/image-pushing/k8s-staging-kubebuilder.yaml (hunk 2 above)
Comment (varshaprasad96) [reply config/jobs/image-pushing/k8s-staging-kubebuilder.yaml:17]: > To build today we call: https://github.com/kubernetes-sigs/kubebuilder/blob/kube-rbac-proxy-releases/build/cloudbuild_kube-rbac-proxy.yaml

To be non-disruptive, do we need to create another cloudbuild.yaml as here? Not entirely sure so keeping the PR as a draft (https://github.com/kubernetes-sigs/kubebuilder/pull/3246). cc: @camilamacedo86

File: config/jobs/image-pushing/k8s-staging-kubebuilder.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            args:
+              - --project=k8s-staging-kubebuilder
+              - --scratch-bucket=gs://k8s-staging-kubebuilder-gcb
+              - --env-passthrough=PULL_BASE_REF
+              - --with-git-dir
+              - .
```
Comment (varshaprasad96) [inline config/jobs/image-pushing/k8s-staging-kubebuilder.yaml:23]: do we need the build-dir?

File: config/jobs/image-pushing/k8s-staging-kubebuilder.yaml (hunk 2 above)
Comment (camilamacedo86) [reply config/jobs/image-pushing/k8s-staging-kubebuilder.yaml:17]: Hi @varshaprasad96,

> To be non-disruptive, do we need to create another cloudbuild.yaml as here? Not entirely sure so keeping the PR as a draft (https://github.com/kubernetes-sigs/kubebuilder/pull/3246). cc: @camilamacedo86

I do not think that we need to have other  cloudbuild.yaml
See that it is only a config file like gorelease one. We can call the same.

However, the new infra should build the images to the new registry
See that we inform the image registry in the build.sh: https://github.com/kubernetes-sigs/kubebuilder/blob/kube-rbac-proxy-releases/build/build.sh#L25

So, we need to change that to receive this info as an ARG OR just create another for the new infra.

File: config/jobs/image-pushing/k8s-staging-kubebuilder.yaml (hunk 1 above)
Comment (camilamacedo86) [reply config/jobs/image-pushing/k8s-staging-kubebuilder.yaml:10]: We do not trigger change anything on the master to build the proxy image. Only its own branch.

File: config/jobs/image-pushing/k8s-staging-kubebuilder.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+postsubmits:
+
+  kubernetes-sigs/kubebuilder:
+    - name: post-kubebuilder-push-images
+      cluster: k8s-infra-prow-build-trusted
+      annotations:
+        testgrid-dashboards: sig-api-machinery-kubebuilder
```
Comment (camilamacedo86) [inline config/jobs/image-pushing/k8s-staging-kubebuilder.yaml:7]: @yashsingh74 from where did you find this value?

Comment (yashsingh74) [reply config/jobs/image-pushing/k8s-staging-kubebuilder.yaml:7]: I find this from here - https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/kubebuilder/kubebuilder-presubmits.yaml#L20

**Pull request closed without merging**